tests/UrlTest.php

```
@@ -30,6 +30,109 @@ public function testUnparseFull()
+        $this->assertEquals($ref, cleanup_url($ref.'?utm_term=1n4l'));
+
+        $this->assertEquals($ref, cleanup_url($ref.'?xtor=some-url'));
+        $this->assertEquals($ref, cleanup_url($ref.'?xtor=some-url&fb=som3th1ng'));
+        $this->assertEquals($ref, cleanup_url(
+            $ref.'?fb=stuff&utm_campaign=zomg&utm_medium=numnum&utm_source=c0d3'
+        ));
+        $this->assertEquals($ref, cleanup_url(
+            $ref.'?xtor=some-url&fb=som3th1ng#tk.rss_all'
+        ));
+
+        // ditch annoying query params and fragment, keep useful params
+        $this->assertEquals(
+            $ref.'?my=stuff&is=kept',
+            cleanup_url(
+                $ref.'?fb=zomg&my=stuff&utm_medium=numnum&is=kept#tk.rss_all'
+            )
+        );
+
+        // ditch annoying query params, keep useful params and fragment
+        $this->assertEquals(
+            $ref.'?my=stuff&is=kept#again',
+            cleanup_url(
+                $ref.'?fb=zomg&my=stuff&utm_medium=numnum&is=kept#again'
+            )
+        );
+    }
+}
+
+/**
+ * Unitary tests for get_url_scheme()
+ */
+class GetUrlSchemeTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * Get empty scheme string for empty Url
+     */
+    public function testGetUrlSchemeEmpty()
+    {
+        $this->assertEquals('', get_url_scheme(''));
+    }
+
+    /**
+     * Get normal scheme of Url
+     */
+    public function testGetUrlScheme()
+    {
+        $url = 'http://domain.tld:3000';
+        $this->assertEquals('http', get_url_scheme($url));
+        $url = 'https://domain.tld:3000';
+        $this->assertEquals('https', get_url_scheme($url));
+        $url = 'domain.tld';
+        $this->assertEquals('http', get_url_shceme($url));
+        $url = 'ssh://domain.tld';
+        $this->assertEquals('ssh', get_url_shceme($url));
+        $url = 'ftp://domain.tld';
+        $this->assertEquals('ftp', get_url_shceme($url));
```

tests should return an error when looking for `get_url_shceme` (typo: `shceme`)

they can be run locally using the Makefile, see [Unit tests](https://github.com/shaarli/Shaarli/wiki/Unit-tests) on the wiki :)

Travis did not run a build for this commit, it happens from time to time... grr